src/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.cs

```diff
@@ -377,30 +374,13 @@ public IAsyncResult BeginSendRequest(DirectoryRequest request, TimeSpan requestT
 
             if (error == 0 && messageID != -1)
             {
-                if (partialMode == PartialResultProcessing.NoPartialResultSupport)
-                {
-                    var requestState = new LdapRequestState();
-                    var asyncResult = new LdapAsyncResult(callback, state, false);
-
-                    requestState._ldapAsync = asyncResult;
-                    asyncResult._resultObject = requestState;
+                // the user registers to retrieve partial results
+                bool partialCallback = partialMode == PartialResultProcessing.ReturnPartialResultsAndNotifyCallback;
 
-                    s_asyncResultTable.Add(asyncResult, messageID);
+                var asyncResult = new LdapPartialAsyncResult(messageID, callback, state, true, this, partialCallback, requestTimeout);
+                s_partialResultsProcessor.Add(asyncResult);
 
-                    _fd.BeginInvoke(messageID, operation, ResultAll.LDAP_MSG_ALL, requestTimeout, true, new AsyncCallback(ResponseCallback), requestState);
-
-                    return asyncResult;
-                }
-                else
```

Comment:
@tquerec, @josephisenhour, @bongiovimatthew-microsoft, do you know the purpose of the above block being deleted?  This change will now cause everything that would have taken the above branch to instead take the below branch, which seems like it would be problematic.  To my knowledge, we don't have particularly good testing of this library in corefx, which would also help to explain why the BeginInvoke usage isn't causing tests to fail.

Reply:
Yes, this doesn't look right to me.  It appears that the change will cause all searches to be processed as "partial".  Partial should only be used by a verify specific type of search, not all searches and is not the default search criteria.  You're also correct that the test coverage for this method is verify basic.  It will only detect major regressions.